src/mito2/src/region/opener.rs

```diff
@@ -427,11 +427,11 @@ pub(crate) async fn replay_memtable<S: LogStore>(
                 .unwrap_or(0);
             region_write_ctx.push_mutation(mutation.op_type, mutation.rows, OptionOutputTx::none());
         }
-    }
 
-    // set next_entry_id and write to memtable.
-    region_write_ctx.set_next_entry_id(last_entry_id + 1);
-    region_write_ctx.write_memtable();
+        // set next_entry_id and write to memtable.
+        region_write_ctx.set_next_entry_id(last_entry_id + 1);
```

Comment:
We should set the entry id outside of the loop.

Reply:
> We should set the entry id outside of the loop.

There's no need to invoke `set_next_entry_id` again outside the loop, as all mutations are written to memtable and their entry ids are updated to version control.